### PR TITLE
Iso upgrade

### PR DESCRIPTION
**Pull request by gitlawr**
https://github.com/rancher/harvester/issues/383
Add support for upgrade via ISO image.

### CODE REVIEW

File: package/upgrade_node.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+# prepare is run before node draining
+prepare()
+{
+  echo "Running prepare scripts"
+  NODE=$(kubectl get po -o=custom-columns=NAME:.metadata.name,NODE:.spec.nodeName|grep $HOSTNAME|awk '{print $2}')
+  # taint the node to trigger VM live migrations
+  kubectl taint node $NODE kubevirt.io/drain=draining:NoSchedule
+  echo "Finished prepare scripts"
+}
```
Comment (guangbochen) [inline package/upgrade_node.sh:39]: do u think we can migrate this script to the controller side to emphasize business logic and infra code separation?

File: pkg/console/constant.go [hunk 2]
```diff
@@ -21,13 +21,16 @@ const (
...
 	spinnerPanel          = "spinner"
+	confirmInstallPanel   = "confirmInstall"
+	confirmUpgradePanel   = "confirmUpgrade"
+	upgradePanel          = "upgrade"
 
-	modeCreate = "create"
-	modeJoin   = "join"
+	modeCreate  = "create"
+	modeJoin    = "join"
+	modeUpgrade = "upgrade"
```
[the remark this answers is not in the document]
Comment (gitlawr) [reply pkg/console/constant.go:33]: Adapted

File: package/upgrade_node.sh (hunk 1 above)
Comment (gitlawr) [reply package/upgrade_node.sh]: Moved

**Comment by gitlawr** [comment]
Updated

**Review by guangbochen — COMMENTED** [review]
@gitlawr can u please rebase this PR, thanks.

**Comment by gitlawr** [comment]
Rebased

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**